tests/webhooks/serviceConversationListener.test.ts

```diff
@@ -0,0 +1,150 @@
+    );
+    helpers.setup({}, runtime);
+  });
+  afterAll(() => {
+    helpers.teardown();
+  });
+  afterEach(() => {
+    mockDocRemove.mockClear();
+  });
+
+  test('Should return status 400 if Body, Media or EventType are undefined', async () => {
+    const event1: Body = {
+      Body: undefined,
+      ConversationSid: 'CHxxxxxxx34EWS',
+      EventType: 'onMessageAdded',
+      Media: {},
+      DateCreated: new Date(),
+    };
+
+    const event2: Body = {
+      Body: 'Test word',
+      ConversationSid: 'CHxxxxxxx34EWS',
+      EventType: 'onMessageAdded',
+      Media: undefined,
+      DateCreated: new Date(),
+    };
+
+    const event3: Body = {
+      Body: 'Test word',
+      ConversationSid: 'CHxxxxxxx34EWS',
+      EventType: undefined,
+      Media: {},
+      DateCreated: new Date(),
+    };
+
+    const callback: ServerlessCallback = (err, result) => {
+      expect(result).toBeDefined();
+      const response = result as MockedResponse;
+
+      expect(response.getStatus()).toBe(400);
+      expect(mockDocRemove).not.toHaveBeenCalled();
+    };
+
+    await serviceConversationListener(baseContext, event1, callback);
+    await serviceConversationListener(baseContext, event2, callback);
+    await serviceConversationListener(baseContext, event3, callback);
+  });
+
+  test('Should return status 500 if object response is undefined', async () => {
+    const event1: Body = {
+      Body: 'Test word',
+      ConversationSid: 'CHxxxxxxx34EWS',
+      EventType: 'onMessageAdded',
+      Media: {},
+      DateCreated: new Date(),
+    };
+
+    const callback1: ServerlessCallback = (err, result) => {
+      expect(result).toBeDefined();
+      const response = result as MockedResponse;
+      expect(response.getStatus()).toBe(400);
+      expect(mockDocRemove).not.toHaveBeenCalled();
+    };
+
+    await serviceConversationListener({ ...baseContext }, event1, callback1);
+
+    const event2: Body = {
+      Body: 'Test word',
+      ConversationSid: 'CHxxxxxxx34EWS',
+      EventType: 'onMessageAdded',
+      Media: {},
+      DateCreated: new Date(),
+    };
+
+    const callback2: ServerlessCallback = (err, result) => {
+      expect(result).toBeDefined();
+      const response = result as MockedResponse;
+      expect(response.getStatus()).toBe(500);
+      expect(mockDocRemove).not.toHaveBeenCalled();
+    };
+
+    await serviceConversationListener(baseContext, event2, callback2);
+  });
+
+  test('Should return status 200', async () => {
+    const event1: Body = {
+      Body: 'Test word',
+      ConversationSid: 'CHxxxxxxx34EWS',
+      EventType: 'onMessageAdded',
+      Media: {},
+      DateCreated: new Date(),
+    };
+
+    const callback1: ServerlessCallback = (err, result) => {
+      expect(result).toBeDefined();
+      const response = result as MockedResponse;
+      expect(response.getStatus()).toBe(200);
+      expect(mockDocRemove).toHaveBeenCalled();
```

Comment:
Could you explain to me how mockDocRemove gets called in this test? I can't see it being passed in or getting set up to mock another method anywhere?

Reply:
@stephenhand - I have removed the `mockDocRemove`. It wasn't doing anything.